orchestrator/lib/az/adb2c/config.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	for name, issuer := range c.ADB2CTrustedIssuers {
+		if name == "" {
+			return errors.New("trusted issuer name cannot be empty")
+		}
+
+		if issuer.IssuerURL == "" {
+			return fmt.Errorf("issuer URL cannot be empty for issuer '%s'", name)
+		}
+
+		if issuer.DiscoveryURL == "" {
+			return fmt.Errorf("discovery URL cannot be empty for issuer '%s'", name)
+		}
+
+		// Validate issuer URL
+		issuerURL, err := url.Parse(issuer.IssuerURL)
+		if err != nil {
+			return fmt.Errorf("invalid issuer URL for issuer '%s': %w", name, err)
+		}
+
+		if issuerURL.Scheme != "https" {
+			return fmt.Errorf("issuer URL must use HTTPS for issuer '%s'", name)
+		}
+
+		// Additional security checks for issuer URL
+		if issuerURL.Host == "" {
+			return fmt.Errorf("issuer URL must have a valid host for issuer '%s'", name)
+		}
+
+		// Validate discovery URL
+		discoveryURL, err := url.Parse(issuer.DiscoveryURL)
+		if err != nil {
+			return fmt.Errorf("invalid discovery URL for issuer '%s': %w", name, err)
+		}
+
+		if discoveryURL.Scheme != "https" {
+			return fmt.Errorf("discovery URL must use HTTPS for issuer '%s'", name)
+		}
+
+		// Additional security checks for discovery URL
+		if discoveryURL.Host == "" {
+			return fmt.Errorf("discovery URL must have a valid host for issuer '%s'", name)
+		}
+
+		// Prevent localhost/private IPs in production (optional - can be configured)
+		if isPrivateOrLocalhost(issuerURL.Host) || isPrivateOrLocalhost(discoveryURL.Host) {
+			// This is a warning-level check - you might want to make this configurable
+			// return fmt.Errorf("private/localhost URLs not allowed in production for issuer '%s'", name)
+		}
+	}
+
+	return nil
+}
+
+// isPrivateOrLocalhost checks if a host is localhost or a private IP
+func isPrivateOrLocalhost(host string) bool {
+	if host == "localhost" || host == "127.0.0.1" || host == "::1" {
+		return true
+	}
+
+	// Check for private IP ranges (10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16)
+	ip := net.ParseIP(host)
+	if ip != nil {
+		return ip.IsLoopback() || ip.IsPrivate()
+	}
+
+	return false
+}
+
+// LoadConfig loads the ADB2C configuration from environment variables using koanf.
+// Environment variables should be prefixed with ADB2C_ and use underscores for nested keys.
+// For example:
+//   - ADB2C_ENABLED=true
+//   - ADB2C_CLIENTID=my-client-id
+//   - ADB2C_TRUSTEDISSUERS_ISSUER1=https://example.com/issuer1
+//   - ADB2C_TRUSTEDISSUERS_ISSUER2=https://example.com/issuer2
+func LoadConfig() (*Config, error) {
+	result := DefaultConfig()
+	err := loadConfigInto(&result)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// DefaultConfig returns the default ADB2C configuration.
+func DefaultConfig() Config {
+	return Config{
+		Enabled:             false,
+		ADB2CClientID:       "",
+		ADB2CTrustedIssuers: make(map[string]TrustedIssuer),
+	}
+}
+
+func loadConfigInto(target any) error {
```

Comment:
why do we need custom config loading for this? Wouldn't you be able to just use whatever map key (that's what we do for the OIDC config), e.g.:

```
ORCA_ADB2C_TRUSTEDISSUERS_0_ISSUERURL=XYZ
ORCA_ADB2C_TRUSTEDISSUERS_0_DISCOVERYURL=XYZ
```
?